docs/advanced-features/using-mdx.md

```diff
@@ -0,0 +1,183 @@
+
+However, because markdown is essentially static content, you can't create _dynamic_ content based on user interactivity. Where MDX shines is in its ability to let you create and use your React components directly in the markup. This opens up a wide range of possibilities when composing your sites pages with interactivity in mind.
+
+## MDX Plugins
+
+Internally MDX uses remark and rehype. Remark is a markdown processor powered by a plugins ecosystem. This plugin ecosystem lets you parse code, transform `HTML` elements, change syntax, extract frontmatter, and more.
+
+Rehype is an `HTML` processor, also powered by a plugin ecosystem. Similar to remark, these plugins let you manipulate, sanitize, compile and configure all types of data, elements and content.
+
+To use a plugin from either remark or rehype, you will need to add it to the MDX packages config.
+
+## `@next/mdx`
+
+The `@next/mdx` package is configured in the `next.config.js` file at your projects root. **It sources data from local files**, allowing you to create pages with a `.mdx` extension, directly in your `/pages` directory.
+
+### Setup `@next/mdx` in Next.js
+
+The following steps outline how to setup `@next/mdx` in your Next.js project:
+
+1. Install the required packages:
+
+   ```terminal
+     npm install @next/mdx @mdx-js/loader
+   ```
+
+2. Require the package and configure to support top level `.mdx` pages. The following adds the `options` object key allowing you to pass in any plugins:
+
+   ```js
+   // next.config.js
+
+   const withMDX = require('@next/mdx')({
+     extension: /\.mdx?$/,
+     options: {
+       remarkPlugins: [],
+       rehypePlugins: [],
+     },
+   })
+   module.exports = withMDX({
+     pageExtensions: ['js', 'jsx', 'md', 'mdx'],
+   })
+   ```
+
+3. Create a new MDX page within the `/pages` directory:
+
+   ```bash
+     - /pages
+       - my-mdx-page.mdx
+     - package.json
+   ```
+
+## Using Components, Layouts and Custom Elements
+
+You can now import a React component directly inside your MDX page. Note that `@next/mdx` does **not** support frontmatter (frontmatter is a YAML like key/value pairing that can be used to store data about a page), instead, you can export data from within the `.mdx` file:
+
+```md
+import { MyComponent } from 'my-components'
+
+export const meta = {
+author: 'Rich Haines'
+}
+
+# My MDX page
+
+This is a list in markdown:
+
+- One
+- Two
+- Three
+
+Checkout my React component:
+
+<MyComponent/>
+```
+
+### Layouts
+
+To add a layout to your MDX page, create a new component and import it into the MDX page. Then you can wrap the MDx page with your layout component:
+
+```md
+import { MyComponent, MyLayoutComponent } from 'my-components'
+
+export const meta = {
+author: 'Rich Haines'
```

Comment:
```suggestion
  author: 'Rich Haines'
```

Should this be indented? I thought lint/prettier would do this automatically?

Reply:
Could question, agreed Prettier should handle this

Reply:
Indeed, prettier didnt do anything to it, so i did it manually